### PR TITLE
Brian/uc 35 merge tables ch

### PR DESCRIPTION
**Pull request by briancao**
Merge session/pageview/event for CH database.

### CODE REVIEW

File: queries/analytics/event/saveEvent.js [hunk 1]
```diff
@@ -33,17 +33,19 @@ async function relationalQuery(website_id, { session_id, url, event_name, event_
...
+
   const params = {
     event_uuid,
     website_id,
-    session_uuid,
     created_at: getDateFormat(new Date()),
     url: url?.substring(0, URL_LENGTH),
     event_name: event_name?.substring(0, EVENT_NAME_LENGTH),
     event_data: JSON.stringify(event_data),
+    ...sessionArgs,
```
Comment (mikecao) [inline queries/analytics/event/saveEvent.js:47]: What are sessionArgs?

File: queries/analytics/pageview/getPageviewMetrics.js [hunk 2]
```diff
@@ -34,26 +34,20 @@ async function relationalQuery(website_id, start_at, end_at, column, table, filt
...
-    and ${getBetweenDates(table + '.created_at', start_at, end_at)}
-    ${pageviewQuery}
-    ${joinSession && sessionQuery}
-    ${eventQuery}
+    from event
+    where website_id= $1
+      ${column !== 'event_name' ? `and event_name = ''` : ''}
+      and ${getBetweenDates('created_at', startDate, endDate)}
+      ${pageviewQuery}
+      ${sessionQuery}
```
Comment (mikecao) [inline queries/analytics/pageview/getPageviewMetrics.js:49]: How does this work without knowing you need to join session?

**Review by mikecao — COMMENTED** [review]
A couple quetsions.

File: queries/analytics/event/saveEvent.js (hunk 1 above)
Comment (briancao) [reply queries/analytics/event/saveEvent.js:47]: we pass the entire session object now, where sessionArgs is the rest of the session data ( hostname, browser, os, device, language).  Country is the only property that needs special care.

File: queries/analytics/pageview/getPageviewMetrics.js (hunk 2 above)
Comment (briancao) [reply queries/analytics/pageview/getPageviewMetrics.js:49]: joinSession only needs to occur if data is collected outside of the primary table being collected on.  Now that everything is merged into 1 table, a join never needs to occur.

File: lib/clickhouse.js [hunk 3]
```diff
@@ -72,48 +72,36 @@ function getFilterQuery(table, column, filters = {}, params = []) {
...
+        arr.push(`and ${key}=$${params.length + 1}`);
+        params.push(decodeURIComponent(filter));
         break;
 
       case 'event_name':
-        if (table === 'event') {
-          arr.push(`and ${table}.${key}=$${params.length + 1}`);
-          params.push(decodeURIComponent(filter));
-        }
+        arr.push(`and ${key}=$${params.length + 1}`);
```
[the remark this answers is not in the document]
Comment (briancao) [reply lib/clickhouse.js:80]: Merged logic.

File: lib/session.js [hunk 4]
```diff
@@ -31,11 +34,11 @@ export async function getSession(req) {
   let websiteId = null;
 
   // Check if website exists
-  if (redis.client) {
+  if (isRedis) {
```
[the remark this answers is not in the document]
Comment (briancao) [reply lib/session.js]: updated language.

**Review by mikecao — APPROVED** [review]

**Pull request merged**